rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -54,7 +54,127 @@ paths:
           $ref: '#/components/responses/NotImplemented'
         default:
           $ref: '#/components/responses/DefaultError'
-
+  /transform/{TransformId}:
+    parameters:
+      - $ref: '#/components/parameters/TransformId'
+    get:
+      summary: Poll transformation request status
+      description: >
+        After posting a transformation request, the client can use the returned URI to poll for the transform's completion at this endpoint.
+      operationId: poll
+      tags:
+        - transform
+      responses:
+        '200':
```

Comment:
❗️ Let's use `100 Continue` instead to indicate this request was not complete as we are still waiting on the final result, at which pint we would return `201` as you have below

Reply:
What if a client does something unexpected with HTTP 100? For HTTP 100, the docs say
`The server, has received the request headers and the client should proceed to send the request body`. Doesn't seem quite like what we want.

Reply:
What about HTTP 102?

Reply:
HTTP 204, no content, seems like it makes sense since.

Reply:
Yeah we saw 102 and didn't like that it's WebDav specific.

In my opinion, none of these status codes really fit this scenario perfectly. My vote is for 204, personally.

Reply:
204 makes sense because we are requesting the temporary download URLs of the generated content, which do not exist at the time of the request.

Reply:
I went with 201 and 204. Unless someone really veto's, I'm going to leave it like that.

Reply:
I wouldn't use a 2xx for this. 2xx means the request is complete and successful. 204 is used to indicate it is complete and there was no need to return anything. This is not the case here. I still prefer to use a 1xx code for this purpose and even though it ain't perfect, 102 remains the best one available. See above comment.

Reply:
@paouelle this all depends on how you look at the request's purpose. To me, the purpose of the request is to check if the transform is done. In the case that it's still in progress, the request has successfully checked if it's done. It's not a failed request.

Reply:
1xx represents informational responses and not errors

Reply:
Let me reword. You said
> 2xx means the request is complete and successful.
What I'm saying is that the request _is_ complete and successful in the case that it finds the transformation is still in progress.

Reply:
On the other end, the request could be seen to get the results from the transformation request which are not yet available and as such, there should be no response which is why, in my opinion, 2xx is not the right thing here. When they are available, we will provide a 200 response with those results. There is no value in adding a in progress status code to the response when you can rely on the HTTP protocol to report that implicitly.

Reply:
I'm not saying we'd add an "in progress" status code to the response. I think you're mixing threads here. I'm saying that we should do `204 No Content` instead of `102`, which by the way, I thought we ruled out because it came from WebDav. To me, `204 No Content` is much better than `100 Continue`, which is the next best option.

Reply:
I am not for a 2xx for that at all. I much prefer sticking with 1xx. Based on the link I provided, I still think 102 might be the best options of what is available regardless of the webdav.

Reply:
@paouelle I just read your link all the way through, and the guy who mentioned 102 said he **did not recommend it** due to some of the constraints around that status code. 102 is not the right answer. The most common status code being used according to all of the below links is to do a 200 for "in progress".
https://stackoverflow.com/questions/5079367/use-http-status-202-for-asynchronous-operations/9527394#9527394
https://farazdagi.com/2014/rest-and-long-running-jobs/
http://restalk-patterns.org/long-running-operation-polling.html